### PR TITLE
71 chore refactor tests

### LINKED ISSUE #71: Chore: Refactor tests
Description: Refactor tests to encapsulate nested fake data inside Fake Data Generator

Technical Details: 

1. Pull back embedded fake data into the parent object.
- Example: generateAccount(accountId, name, etc.) --> generateAccount(). Under this theory, generateAccount would generate all of its needed attributes within the method as opposed to passing them in to the method itself.
2. Apply setters to Model objects as necessary 
- Example PetId, AccountId, etc.

Acceptance Criteria:
- Refactor all necessary tests
- Meet DoD

Depends on #59 and #69 being completed first.

### PR DESCRIPTION
**Pull request by jsnorwood**
Added comments for where tests should be refactored.

### CODE REVIEW

File: src/test/java/com/VetApp/PurrgentCare/FakeDataGenerator.java [hunk 1]
```diff
@@ -93,11 +123,19 @@ public List<PersonResponse> generateFakePersonResponseList(Integer countOfFakePe
     //Account Types
     /////
 
-    public Account generateDefaultAccount() {
+    public Account generateFakeAccount() {
         return Instancio.of(Account.class).create();
     }
 
-    public List<Account> generateDefaultAccountList() {
+    public Account generateFakeAccount(Account fakeAccount) {
```
Comment (RandallKeur) [inline src/test/java/com/VetApp/PurrgentCare/FakeDataGenerator.java:130]: Could use javadoc similar to Pet and Person overloaded methods.

File: src/test/java/com/VetApp/PurrgentCare/services/AccountServiceImplementationTest.java [hunk 2]
```diff
@@ -67,22 +69,28 @@ public void getAccount_whenExist_returnOneAccount() {
...
-        final var fakeAccountId = fakeDataGenerator.generateRandomInteger();
-        final var emptyAccountResponse = new AccountResponse();
+        final var fakeAccount = fakeDataGenerator.generateFakeAccount();
+        final var fakeAccountId = fakeAccount.getId();
+        final var fakeAccountResponse = fakeDataGenerator.generateFakeAccountResponse();
+        fakeAccountResponse.setId(null);
+        fakeAccountResponse.setActive(null);
+        fakeAccountResponse.setDateCreated(null);
+        fakeAccountResponse.setPets(null);
+        fakeAccountResponse.setAccountHolders(null);
```
Comment (RandallKeur) [inline src/test/java/com/VetApp/PurrgentCare/services/AccountServiceImplementationTest.java:79]: could we make this a separate method in the FakeDataGenerator?
Something like `generateEmptyAccountResponse`?

File: src/test/java/com/VetApp/PurrgentCare/services/AccountServiceImplementationTest.java [hunk 3]
```diff
@@ -169,10 +179,11 @@ public void updateAccount_whenAccountExists_returnsUpdatedAccount() {
     }
 
     @Test
+
     public void updateAccount_whenAccountNotExists_throwEntityNotFoundException() {
         // given
         final var fakeAccountRequest = fakeDataGenerator.generateFakeAccountRequest();
-        final var fakeAccountId = fakeDataGenerator.generateRandomInteger();
+        final var fakeAccountId = fakeDataGenerator.generateFakeAccount().getId();
```
Comment (RandallKeur) [inline src/test/java/com/VetApp/PurrgentCare/services/AccountServiceImplementationTest.java:186]: Did we need to change this?
Do we need a fakeAccount object?

File: src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java [hunk 4]
```diff
@@ -63,11 +63,19 @@ public void getPet_whenExist_returnOnePet() {
...
-        final var fakePetId = fakeDataGenerator.generateRandomInteger();
-        final var fakePetResponse = new PetResponse();
+        final var fakePet = fakeDataGenerator.generateFakePet();
+        final var fakePetId = fakePet.getId();
+        final var fakePetResponse = fakeDataGenerator.generateFakePetResponse();
+        fakePetResponse.setId(null);
+        fakePetResponse.setName(null);
+        fakePetResponse.setType(null);
+        fakePetResponse.setAge(null);
+        fakePetResponse.setGender(null);
```
Comment (RandallKeur) [inline src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java:73]: Similar to the Account comment, is there a way we could either make a method to generate a fake pet response or initialize this in the test to be an empty response?

File: src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java [hunk 5]
```diff
@@ -146,38 +153,35 @@ public void getAllPets_whenNoPets_returnsEmptyList() {
...
                 .isEqualTo(fakePetResponse);
+
     }
 
     @Test
     public void updatePet_whenPetNotExists_throwEntityNotFoundException() {
         // given
         final var fakePetRequest = fakeDataGenerator.generateFakePetRequest();
-        final var fakePetId = fakeDataGenerator.generateRandomInteger();
+        final var fakePetId = fakeDataGenerator.generateFakePet().getId();
```
Comment (RandallKeur) [inline src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java:184]: is it necessary to have the fake pet object?
Was this change necessary?

File: src/test/java/com/VetApp/PurrgentCare/FakeDataGenerator.java (hunk 1 above)
Comment (jsnorwood) [reply src/test/java/com/VetApp/PurrgentCare/FakeDataGenerator.java:130]: Aren't we already doing an overloaded method for generateFakeAccount?

File: src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java (hunk 5 above)
Comment (jsnorwood) [reply src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java:184]: It's sort of an apples to apples change. Basically doing the same thing so not _necessarily_ critical to change.

**Review by RandallKeur — DISMISSED** [review]

**Review by RandallKeur — APPROVED** [review]
